File: 55. Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_achieve = 0;
        for (int i = 0; i < nums.size()-1; i++)
        {
            max_achieve = max_achieve > i + nums[i] ? max_achieve : i + nums[i];
            if (max_achieve < i+1)
                return false;
        }
        return true;
    }
};

File: 55. Jump Game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        max_achieve = 0
        for i in xrange(len(nums)-1):
            max_achieve = max(max_achieve, i + nums[i])
            if max_achieve < i+1:
                return False
        return True
